src/aliases/api/handlers/handlers.go

```diff
@@ -0,0 +1,297 @@
+func (h *AliasHandler) getAlias(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+	key := vars["alias_key"]
+
+	err := validatePathVars(regName, key)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	alias, err := h.alias.GetAlias(r.Context(), aliasent.RegistryName(regName), aliasent.AliasKey(key))
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+
+	err = infrahttp.WriteJSON(w, types.AliasResponse{
+		Value: types.AliasValue(alias.Value),
+	})
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+}
+
+// updateAlias updates an alias value.
+// @Summary Update an alias
+// @Description Update an alias of a key from a dedicated alias registry
+// @Tags alias
+// @Accept json
+// @Produce json
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Param request body types.AliasRequest true "Update Alias Request"
+// @Success 200 {object} types.AliasResponse "Alias data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 404 {object} ErrorResponse "Alias not found"
+// @Failure 500 {object} ErrorResponse "Internal server error"
+// @Router /registries/{registry_name}/aliases/{alias_key} [put]
+func (h *AliasHandler) updateAlias(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+	key := vars["alias_key"]
+
+	err := validatePathVars(regName, key)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	var aliasReq types.AliasRequest
+	err = jsonutils.UnmarshalBody(r.Body, &aliasReq)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	alias := &aliasent.Alias{
+		RegistryName: aliasent.RegistryName(regName),
+		Key:          aliasent.AliasKey(key),
+		Value:        aliasent.AliasValue(aliasReq.Value),
+	}
+
+	alias, err = h.alias.UpdateAlias(r.Context(), aliasent.RegistryName(regName), *alias)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+
+	err = infrahttp.WriteJSON(w, types.AliasResponse{
+		Value: types.AliasValue(alias.Value),
+	})
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+}
+
+// deleteAlias deletes an alias value.
+// @Summary Delete an alias
+// @Description Delete an alias of a key from a dedicated alias registry
+// @Tags alias
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Success 204 "Deleted successfully"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 404 {object} ErrorResponse "Alias not found"
```

Comment:
In case of registry not found, isn't it a 404 too?

Reply:
We actually don't have a registry, it's part of the same table, and it's the same composed key. So either/or, it will fail.
And since adding a key to an unexisting registry will create the "registry", then it will succeed.
It can either succeed if both ids are good, or it fails if any of them fails, we can't know which was wrong. But a good alias key in the wrong registry doesn't make it good anyway. So just a 404 saying about Alias should be good.

The real response says the resource is unavailable.
```
{"message":"ST100: resource not found","code":"ST100"}
```